### PR TITLE
SONARIAC-1154 Resolve values locations in values.yaml

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: sonar-helm-for-iac/converters/tree_converter.go [hunk 1]
```diff
@@ -209,7 +209,7 @@ func (c ConversionContext) convertFieldNode(node parse.FieldNode) proto.Message
 	return &pbstructs.FieldNode{
 		NodeType: pbstructs.NodeType_NodeField,
 		Pos:      int64(node.Position()),
-		Length:   int64(len(node.Ident)),
+		Length:   nodeSourceLength(&node),
```
Comment (petertrr) [inline sonar-helm-for-iac/converters/tree_converter.go:212]: Overlooked it during the initial implementation

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java [hunk 2]
```diff
@@ -20,67 +20,167 @@
...
+    ListNodeImpl root = new ListNodeImpl(0, 15, List.of(textNode, actionNode));
+    var goTemplateTree = new GoTemplateTreeImpl("test", "test", 0, root);
+    inputFileContext.setGoTemplateTree(goTemplateTree);
+
+    var adjustableCheckContext = (AdjustableChecksVisitor.AdjustableContextAdapter) adjustableChecksVisitor.context(null);
+
+    var locationsInAdditionalFiles = adjustableCheckContext.findSecondaryLocationsInAdditionalFiles(inputFileContext, TextRanges.range(1, 6, 1, 22));
+
+    Assertions.assertThat(locationsInAdditionalFiles).hasSize(1);
+    TextRangeAssert.assertThat(locationsInAdditionalFiles.get(0).textRange).hasRange(1, 5, 1, 8);
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java:81]: static imports

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java [hunk 3]
```diff
@@ -20,67 +20,167 @@
...
+    var inputFileContext = new HelmInputFileContext(mockSensorContextWithEnabledFeature(), null);
+    inputFileContext.setAdditionalFiles(Map.of());
 
-  @Test
-  void shouldReturnNullForMissingPath() throws IOException {
-    var textRange = getTextRangeFor(code(
-      "foo:",
-      "  bar:",
-      "    baz: qux"), new ValuePath("foo", "baz"));
+    var locations = adjustableCheckContext.maybeFindSecondaryLocationsInAdditionalFiles(inputFileContext, null);
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java:166]: Please rename to `secondaryLocations`

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java [hunk 4]
```diff
@@ -20,67 +20,167 @@
...
+
+    var adjustableCheckContext = (AdjustableChecksVisitor.AdjustableContextAdapter) adjustableChecksVisitor.context(null);
+
+    var locationsInAdditionalFiles = adjustableCheckContext.findSecondaryLocationsInAdditionalFiles(inputFileContext, TextRanges.range(1, 6, 1, 22));
+
+    Assertions.assertThat(locationsInAdditionalFiles).hasSize(1);
+    TextRangeAssert.assertThat(locationsInAdditionalFiles.get(0).textRange).hasRange(1, 5, 1, 8);
+
+    locationsInAdditionalFiles = adjustableCheckContext.findSecondaryLocationsInAdditionalFiles(inputFileContext, TextRanges.range(1, 1, 1, 3));
+    Assertions.assertThat(locationsInAdditionalFiles).isEmpty();
```
[the remark this answers is not in the document]
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java:84]: Actually, I see 3 calls of `findSecondaryLocationsInAdditionalFiles` so I will suggest to split it to 3 different test cases

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java [hunk 5]
```diff
@@ -20,67 +20,167 @@
...
+import static org.sonar.iac.common.testing.IacTestUtils.inputFile;
 
 class AdjustableChecksVisitorTest {
   @Test
-  void shouldFindValueInValuesFile() throws IOException {
-    var textRange = getTextRangeFor(code(
-      "foo:",
-      "  bar:",
-      "    baz: qux"), new ValuePath("foo", "bar", "baz"));
+  void shouldFindSecondaryLocationsInValuesFile() throws IOException {
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java:58]: `throws IOException` is not needed

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java [hunk 6]
```diff
@@ -20,67 +20,167 @@
...
+        "  bar:",
+        "    - baz: qux"), new ValuePath("foo", "bar", "baz")),
+      Arguments.of("foo: bar", new ValuePath("Chart", "Name")),
+      Arguments.of("foo: bar", new ValuePath("Release", "Name")),
+      Arguments.of("[1, 2] : foo", new ValuePath("Values", "bar")));
   }
 
   @Test
   void shouldReturnNullForMissingValuesFile() throws IOException {
     var adjustableChecksVisitor = new AdjustableChecksVisitor(Mockito.mock(Checks.class), null, null, new YamlParser());
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitorTest.java:149]: static import of `Mockito.mock`

File: sonar-helm-for-iac/converters/tree_converter.go (hunk 1 above)
Comment (mstachniuk) [reply sonar-helm-for-iac/converters/tree_converter.go:212]: No test needs changes due to this modification? Is it tested somewhere?

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java [hunk 7]
```diff
@@ -78,45 +89,101 @@ public <T extends Tree> void register(Class<T> cls, BiConsumer<CheckContext, T>
     @Override
     protected void reportIssue(@Nullable TextRange textRange, String message, List<SecondaryLocation> secondaryLocations) {
       var shiftedTextRange = textRange;
+      List<SecondaryLocation> enhancedAndAdjustedSecondaryLocations = new ArrayList<>();
       if (textRange != null) {
         shiftedTextRange = locationShifter.computeShiftedLocation(currentCtx, textRange);
+
+        enhancedAndAdjustedSecondaryLocations = maybeFindSecondaryLocationsInAdditionalFiles(currentCtx, shiftedTextRange);
+      }
+      enhancedAndAdjustedSecondaryLocations.addAll(secondaryLocations.stream().map(this::adaptSecondaryLocation).collect(Collectors.toList()));
```
[the remark this answers is not in the document]
Comment (petertrr) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java:98]: These secondary locations are from the same file

**Pull request merged**